### PR TITLE
Add all deterministic encryption functions

### PR DESCRIPTION
**Pull request by giarc3**

### CODE REVIEW

**Review by skeet70 — APPROVED** [review]
Worked for me. I do think we should add a comment about why we have this with a link to relevant github issues so we might be have the knowledge to remove it in the future.

File: src/Util.ts [hunk 1]
```diff
@@ -16,6 +18,17 @@ const parseErrorFromFailedResponse = (failureResponse: Response) =>
         .errorMap(() => new TspServiceException(TenantSecurityErrorCode.UNKNOWN_ERROR, "Unknown response from Tenant Security Proxy", failureResponse.status))
         .flatMap((errorResp: ApiErrorResponse) => Future.reject(TenantSecurityExceptionUtils.from(errorResp.code, errorResp.message, failureResponse.status)));
 
+const httpAgent = new http.Agent({keepAlive: true});
+const httpsAgent = new https.Agent({keepAlive: true});
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const agentSelector = function (_parsedURL: any) {
+    if (_parsedURL.protocol == "http:") {
```
Comment (coltfred) [inline src/Util.ts:25]: Why the leading `_`?

File: src/Util.ts [hunk 2]
```diff
@@ -16,6 +18,17 @@ const parseErrorFromFailedResponse = (failureResponse: Response) =>
         .errorMap(() => new TspServiceException(TenantSecurityErrorCode.UNKNOWN_ERROR, "Unknown response from Tenant Security Proxy", failureResponse.status))
         .flatMap((errorResp: ApiErrorResponse) => Future.reject(TenantSecurityExceptionUtils.from(errorResp.code, errorResp.message, failureResponse.status)));
 
+const httpAgent = new http.Agent({keepAlive: true});
+const httpsAgent = new https.Agent({keepAlive: true});
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const agentSelector = function (_parsedURL: any) {
```
Comment (coltfred) [inline src/Util.ts:24]: Should we comment on why this is needed?

**Review by coltfred — COMMENTED** [review]
Just a couple questions.

File: src/Util.ts (hunk 1 above)
Comment (giarc3) [reply src/Util.ts:28]: Not sure, probably some convention I don't know. I think it works either way, but this is how I copied it.

File: src/Util.ts (hunk 2 above)
Comment (giarc3) [reply src/Util.ts:27]: Yep, added

**Pull request merged**